### DISCUSSION TITLE
posts/2022/04/gems-in-rough-17/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#8] [Announcements]
# posts/2022/04/gems-in-rough-17/

A GitHub Discussions oddity, Netlify Edge Functions, feed readers with built-in browsers.

https://www.brycewray.com/posts/2022/04/gems-in-rough-17/

**Comment by davidsneighbour** [comment]
> ...where they should be eternally safe from my fickleness.

Sitting here with my popcorn ready for the moment you migrate to an engine that uses weird individual links ;) In that case my advise would be to start using the title of the posts as indicator for the discussion. Looks also better than the permalink in the discussion repository.

**Reply by brycewray** [reply]
Oh, I wouldn’t use any platform that requires me to change my posts’ URLs. While I could use rewrite rules to get around that (mostly), I would assume there would be some “search juice” cost to doing that.

Still, I’ll take yet another look at the giscus documentation regarding such options. I’m not concerned with how they look in the GH Discussions UI itself, but it nonetheless would be good to re-consider my options.

**Reply by davidsneighbour** [reply]
The rewriting would not rewrite the title of your discussion is what I am meaning. But you can always go via Githubs REST API to rewrite them to their new counterpart. The option for giscus is very hidden and not really documented, but if you go on the [giscus website](https://giscus.app/) and check under "Page ↔️ Discussions Mapping", then check in the prepared code. The `data-mapping` parameter has it.

**Reply by brycewray** [reply]
Yes, saw that. So, for existing ones (like this one now), I’d have to go back and change the Discussion titles, correct? Otherwise giscus would no longer connect the existing Discussions with the pages from which they originated.

**Edit**: disregard; that’s exactly what you were saying above. Sorry; too little sleep is my excuse, and I hope it will suffice.

**Further edit**: have now done. Thanks for the suggestion!

**Still further edit**: @davidsneighbour I had to go back to pathnames, because giscus got confused on a couple of them that are similar (specifically, was putting the comments from **this** page on the page for the original “Gems in the rough” as well as on this page).

**Even still further edit**: Turns out the aforementioned title-confusion glitch is a [known giscus issue](https://github.com/giscus/giscus/issues/508).

**And yet another**: The 2022-07-23 fix to that known giscus issue involves strict matching with a hash of the chosen item, so I (theoretically) could go back and change this accordingly, now. That said, the matching is **so** strict that I think I’m better off sticking with pathnames, after all (long story but, trust me, I checked).

**Reply by sarah11918** [reply]
Woo! I am now able to comment, and I'm glad if I could have helped in any small way here. Lovely to have you back as a regular in the Astro community!

**Reply by brycewray** [reply]
Very glad to be back! Yes, you definitely did help me get this one resolved. Indeed, all of you helped — and continue to help — a **great** deal with my various Astro-related undertakings. Even today, I got help from Chris with a particularly nasty problem I was have getting a style to work because it was one of those `set:html` things, and therefore Vite wasn’t picking it up; but he showed me how. Finally, thanks to the release of 1.0.0-beta.20 and the fix for the `define:vars` bug, my posts’ hero images even have their Base64 LQIPs back! ✅ ✅ ✅ 🎉 🎉 🎉
